unstructured_ingest/v2/processes/connectors/vectara.py

```diff
@@ -0,0 +1,310 @@
+import json
+import uuid
+from dataclasses import dataclass, field
+from datetime import date, datetime
+from pathlib import Path
+from typing import Any, Dict, List, Mapping, Optional
+
+from dateutil import parser
+from pydantic import Field
+
+from unstructured_ingest.enhanced_dataclass import enhanced_field
+from unstructured_ingest.error import DestinationConnectionError
+from unstructured_ingest.utils.data_prep import flatten_dict
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    FileData,
+    Uploader,
+    UploaderConfig,
+    UploadStager,
+    UploadStagerConfig,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import DestinationRegistryEntry
+
+BASE_URL = "https://api.vectara.io/v1"
+
+CONNECTOR_TYPE = "vectara"
+
+
+class VectaraAccessConfig(AccessConfig):
+    oauth_client_id: str = enhanced_field(sensitive=True)
```

Comment:
Let's get descriptions in all these class attributes. See:
https://github.com/Unstructured-IO/unstructured-ingest/blob/b3add2abf347b5e21121bf01626f8b4e9e3ec9e1/unstructured_ingest/v2/processes/connectors/elasticsearch.py#L49

Since we are no longer using the CLI tool Click and are using Pydantic for our CLI help descriptions.